### PR TITLE
Generated E2E upgrade scenario

### LINKED ISSUE #2039: Implement upgrade scenario

### PR DESCRIPTION
**Pull request by kron4eg**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2039 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Generated E2E upgrade scenario
```

### CODE REVIEW

File: testv2/e2e/scenario_upgrade.go [hunk 1]
```diff
@@ -45,24 +46,51 @@ func (scenario *scenarioUpgrade) SetVersions(versions ...string) {
...
+		t.Fatalf("kubeone apply failed: %v", err)
+	}
 }
 
 func (scenario *scenarioUpgrade) test(t *testing.T) {
-	// TODO: add some testings
+	t.Helper()
+
+	data := manifestData{
+		VERSION: scenario.versions[1],
```
Comment (xmudrii) [inline testv2/e2e/scenario_upgrade.go:83]: Should this eventually use the first version (i.e. `versions[0]`)?

**Review by xmudrii — COMMENTED** [review]
Just a question, otherwise LGTM.

File: testv2/e2e/scenario_upgrade.go (hunk 1 above)
Comment (kron4eg) [reply testv2/e2e/scenario_upgrade.go:83]: So if we take for example 1 upgrade test
```
func TestAwsDefaultsUpgradeContainerdFromV1_21_12_ToV1_22_9(t *testing.T) {
        infra := Infrastructures["aws_defaults"]
        scenario := Scenarios["upgrade_containerd"]
        scenario.SetInfra(infra)
        scenario.SetVersions("v1.21.12", "v1.22.9")
        scenario.Run(t)
}
```

we can se that versions[0] is OLD version, and versions[1] is NEW version. Obviously the versions[1] it used to verify the cluster after upgrade has been done.

Comment (xmudrii) [reply testv2/e2e/scenario_upgrade.go:83]: Oh, I see the `run` function now. Okay, it makes sense now.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve :zap:

**Pull request merged**